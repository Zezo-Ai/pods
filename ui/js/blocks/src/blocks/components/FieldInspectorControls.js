/**
 * WordPress dependencies
 */
import { PanelRow } from '@wordpress/components';

/**
 * Internal dependencies
 */
import RenderedField from './RenderedField';

/**
 * Renders the fields that live in the Inspector on the sidebar.
 *
 * @param {Object} root0
 * @param {Array} root0.fields
 * @param {Object} root0.attributes
 * @param {Object} root0.setAttributes
 */
const FieldInspectorControls = ( {
	fields = [],
	attributes,
	setAttributes,
} ) => {
	if ( ! fields.length ) {
		return null;
	}

	return (
		<div className="pods-inspector-rows" data-testid="pods-inspector">
			{ fields.map( ( field ) => {
				const {
					name,
				} = field;

				return (
					<PanelRow key={ name } className="pods-inspector-row">
						<RenderedField
							field={ field }
							attributes={ attributes }
							setAttributes={ setAttributes }
						/>
					</PanelRow>
				);
			} ) }
		</div>
	);
};

export default FieldInspectorControls;
