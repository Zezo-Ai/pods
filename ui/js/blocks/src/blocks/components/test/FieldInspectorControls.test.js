/**
 * External dependencies
 */
import { render, screen } from '@testing-library/react';

/**
 * WordPress dependencies
 */
import { RichText } from '@wordpress/block-editor';

import {
	CheckboxControl,
	DateTimePicker,
	TextControl,
	TextareaControl,
	RadioControl,
} from '@wordpress/components';

/**
 * Internal dependencies
 */
import FieldInspectorControls from '../FieldInspectorControls';
import CheckboxGroup from '../../../components/CheckboxGroup';
import NumberControl from '../../../components/NumberControl';

import {
	allFieldsBlock,
	allFieldsBlockProps,
} from '../../../testData';

const allFieldsEditComponentProps = {
	...allFieldsBlockProps,
	setAttributes: jest.fn(),
};

window.ResizeObserver =
	window.ResizeObserver ||
	jest.fn().mockImplementation(() => ({
		disconnect: jest.fn(),
		observe: jest.fn(),
		unobserve: jest.fn(),
	}));

describe( 'FieldInspectorControls', () => {
	test( 'all supported fields render as expected', () => {
		const { fields } = allFieldsBlock;

		const {
			attributes,
			setAttributes,
		} = allFieldsEditComponentProps;

		render(
			<FieldInspectorControls
				fields={ fields }
				attributes={ attributes }
				setAttributes={ setAttributes }
			/>
		);

		expect( screen.getByTestId( 'pods-inspector' ).tagName ).toEqual( 'DIV' );
	} );
} );
