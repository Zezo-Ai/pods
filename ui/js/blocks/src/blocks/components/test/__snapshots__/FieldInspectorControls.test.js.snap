// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldInspectorControls all supported fields render as expected 1`] = `"<div class="components-panel__row pods-inspector-row"><div class="components-base-control css-fothu4-Wrapper-boxSizingReset ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><label class="components-base-control__label css-1jylnrs-StyledLabel-baseLabelTypography-labelStyles ej5x27r2" for="inspector-text-control-0">Label for the text field</label><input class="components-text-control__input" type="text" id="inspector-text-control-0" aria-describedby="inspector-text-control-0__help" value="<em>Content for the text field</em>, but these <script></script>tags will be stripped <strong>out</strong>"></div><p id="inspector-text-control-0__help" class="components-base-control__help css-shdw93-StyledHelp ej5x27r1">Some help text</p></div></div><div class="components-panel__row pods-inspector-row"><div class="components-base-control css-fothu4-Wrapper-boxSizingReset ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><label class="components-base-control__label css-1jylnrs-StyledLabel-baseLabelTypography-labelStyles ej5x27r2" for="inspector-textarea-control-0">Label for the textarea field</label><textarea class="components-textarea-control__input css-13tehp6-StyledTextarea-inputStyleNeutral-inputStyleFocus e1w5nnrk0" id="inspector-textarea-control-0" rows="4" aria-describedby="inspector-textarea-control-0__help">&lt;em&gt;Content for the textarea field&lt;/em&gt;, but these &lt;script&gt;&lt;/script&gt;tags will be stripped &lt;strong&gt;out&lt;/strong&gt;</textarea></div><p id="inspector-textarea-control-0__help" class="components-base-control__help css-shdw93-StyledHelp ej5x27r1">Some help text</p></div></div><div class="components-panel__row pods-inspector-row"><p role="textbox" aria-multiline="true" aria-readonly="false" contenteditable="true" class="block-editor-rich-text__editable rich-text" style="white-space: pre-wrap; min-width: 1px;">Some content</p></div><div class="components-panel__row pods-inspector-row"><fieldset class="components-block-fields-checkbox-control"><legend>Checkbox Field</legend><div class="components-base-control components-checkbox-control css-fothu4-Wrapper-boxSizingReset ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><div data-wp-c16t="true" data-wp-component="HStack" class="components-flex components-h-stack css-1m5h1bg-PolymorphicDiv-Flex-base-ItemsRow e19lxcc00"><span class="components-checkbox-control__input-container"><input id="inspector-checkbox-control-0" class="components-checkbox-control__input" type="checkbox" aria-describedby="inspector-checkbox-control-0__help" value="1" checked=""><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24" role="presentation" class="components-checkbox-control__checked" aria-hidden="true" focusable="false"><path d="M16.7 7.1l-6.3 8.5-3.3-2.5-.9 1.2 4.5 3.4L17.9 8z"></path></svg></span><label class="components-checkbox-control__label" for="inspector-checkbox-control-0">Checkbox Test</label></div></div><p id="inspector-checkbox-control-0__help" class="components-base-control__help css-shdw93-StyledHelp ej5x27r1"><span class="components-checkbox-control__help">Additional help text</span></p></div></fieldset></div><div class="components-panel__row pods-inspector-row"><fieldset class="components-block-fields-checkbox-group"><legend>Checkbox Field</legend><div class="components-base-control components-checkbox-control css-fothu4-Wrapper-boxSizingReset ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><div data-wp-c16t="true" data-wp-component="HStack" class="components-flex components-h-stack css-1m5h1bg-PolymorphicDiv-Flex-base-ItemsRow e19lxcc00"><span class="components-checkbox-control__input-container"><input id="inspector-checkbox-control-1" class="components-checkbox-control__input" type="checkbox" value="1" checked=""><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24" role="presentation" class="components-checkbox-control__checked" aria-hidden="true" focusable="false"><path d="M16.7 7.1l-6.3 8.5-3.3-2.5-.9 1.2 4.5 3.4L17.9 8z"></path></svg></span><label class="components-checkbox-control__label" for="inspector-checkbox-control-1">First Option</label></div></div></div><div class="components-base-control components-checkbox-control css-fothu4-Wrapper-boxSizingReset ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><div data-wp-c16t="true" data-wp-component="HStack" class="components-flex components-h-stack css-1m5h1bg-PolymorphicDiv-Flex-base-ItemsRow e19lxcc00"><span class="components-checkbox-control__input-container"><input id="inspector-checkbox-control-2" class="components-checkbox-control__input" type="checkbox" value="1" checked=""><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24" role="presentation" class="components-checkbox-control__checked" aria-hidden="true" focusable="false"><path d="M16.7 7.1l-6.3 8.5-3.3-2.5-.9 1.2 4.5 3.4L17.9 8z"></path></svg></span><label class="components-checkbox-control__label" for="inspector-checkbox-control-2">Second Option</label></div></div></div><div class="components-base-control components-checkbox-control css-fothu4-Wrapper-boxSizingReset ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><div data-wp-c16t="true" data-wp-component="HStack" class="components-flex components-h-stack css-1m5h1bg-PolymorphicDiv-Flex-base-ItemsRow e19lxcc00"><span class="components-checkbox-control__input-container"><input id="inspector-checkbox-control-3" class="components-checkbox-control__input" type="checkbox" value="1"></span><label class="components-checkbox-control__label" for="inspector-checkbox-control-3">Third Option</label></div></div></div><p id="__help" class="components-block-fields-checkbox-group__help">Additional help text</p></fieldset></div><div class="components-panel__row pods-inspector-row"><fieldset id="inspector-radio-control-0" class="components-radio-control" aria-describedby="inspector-radio-control-0__help"><legend class="components-base-control__label css-1w2n0ib-StyledVisualLabel-baseLabelTypography-labelStyles ej5x27r0"></legend><div data-wp-c16t="true" data-wp-component="VStack" class="components-flex components-h-stack components-v-stack components-radio-control__group-wrapper has-help css-1ykmsse-PolymorphicDiv-Flex-base-ItemsColumn e19lxcc00"><div class="components-radio-control__option"><input id="inspector-radio-control-0-0" class="components-radio-control__input" type="radio" name="inspector-radio-control-0" value="first" checked=""><label class="components-radio-control__label" for="inspector-radio-control-0-0">First Option</label></div><div class="components-radio-control__option"><input id="inspector-radio-control-0-1" class="components-radio-control__input" type="radio" name="inspector-radio-control-0" value="second"><label class="components-radio-control__label" for="inspector-radio-control-0-1">Second Option</label></div><div class="components-radio-control__option"><input id="inspector-radio-control-0-2" class="components-radio-control__input" type="radio" name="inspector-radio-control-0" value="third"><label class="components-radio-control__label" for="inspector-radio-control-0-2">Third Option</label></div></div><p id="inspector-radio-control-0__help" class="components-base-control__help css-shdw93-StyledHelp ej5x27r1">Additional help text</p></fieldset></div><div class="components-panel__row pods-inspector-row"><div class="components-base-control full-width-base-control css-fothu4-Wrapper-boxSizingReset ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><div class="css-fyq6mk-container" id="inspector-select-control-0"><span id="react-select-2-live-region" class="css-1f43avz-a11yText-A11yText"></span><span aria-live="polite" aria-atomic="false" aria-relevant="additions text" role="log" class="css-1f43avz-a11yText-A11yText"></span><div class="css-13cymwt-control"><div class="css-hlgwow"><div class="css-1jqq78o-placeholder" id="react-select-2-placeholder">Select...</div><div class="css-19bb58m" data-value=""><input class="" style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;" autocapitalize="none" autocomplete="off" autocorrect="off" id="react-select-2-input" spellcheck="false" tabindex="0" type="text" aria-autocomplete="list" aria-expanded="false" aria-haspopup="true" role="combobox" aria-activedescendant="" aria-describedby="react-select-2-placeholder" value=""></div></div><div class="css-1wy0on6"><span class="css-1u9des2-indicatorSeparator"></span><div class="css-1xc3v61-indicatorContainer" aria-hidden="true"><svg height="20" width="20" viewBox="0 0 20 20" aria-hidden="true" focusable="false" class="css-tj5bde-Svg"><path d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"></path></svg></div></div></div><input name="selectField" type="hidden" value=""></div></div></div></div><div class="components-panel__row pods-inspector-row"><div class="components-base-control css-fothu4-Wrapper-boxSizingReset ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><span class="components-base-control__label css-1w2n0ib-StyledVisualLabel-baseLabelTypography-labelStyles ej5x27r0">Label for the datetime field</span><div data-wp-c16t="true" data-wp-component="VStack" class="components-flex components-h-stack components-v-stack components-datetime e1p5onf00 css-1f5nl9f-PolymorphicDiv-Flex-base-ItemsColumn-Wrapper e19lxcc00"><div class="components-datetime__time css-q9jb5p-Wrapper evcr2319"><fieldset class="css-1jlauhr-Fieldset evcr2318"><legend class="components-base-control__label components-datetime__time-legend css-1w2n0ib-StyledVisualLabel-baseLabelTypography-labelStyles ej5x27r0">Time</legend><div data-wp-c16t="true" data-wp-component="HStack" class="components-flex components-h-stack components-datetime__time-wrapper css-1897a1g-PolymorphicDiv-Flex-base-ItemsRow e19lxcc00"><div data-wp-c16t="true" data-wp-component="HStack" class="components-flex components-h-stack css-14sym1v-PolymorphicDiv-Flex-base-ItemsRow e19lxcc00"><div class="components-datetime__time-field components-datetime__time-field-time css-tzgm3n-TimeWrapper evcr2317"><div class="components-base-control components-input-control components-number-control components-datetime__time-field-hours-input evcr2316 ep09it41 css-ztoleg-Wrapper-boxSizingReset-Input-htmlArrowStyles-HoursInput-baseInput ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><div data-wp-c16t="true" data-wp-component="InputBase" class="components-flex components-input-base em5sgkm5 css-y3fmmm-PolymorphicDiv-Flex-base-ItemsColumn-Root e19lxcc00"><label data-wp-c16t="true" data-wp-component="VisuallyHidden" for="inspector-input-control-0" class="components-visually-hidden css-1qfl78b-PolymorphicDiv e19lxcc00" style="border: 0px; clip: rect(1px, 1px, 1px, 1px); clip-path: inset( 50% ); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; word-wrap: normal;">Hours</label><div class="components-input-control__container css-7xxir9-Container-containerDisabledStyles-containerWidthStyles em5sgkm4"><input autocomplete="off" inputmode="numeric" max="12" min="1" required="" step="1" class="components-input-control__input css-1chjf1n-Input-dragStyles-fontSizeStyles-sizeStyles-customPaddings em5sgkm3" id="inspector-input-control-0" type="number" value="11"><div aria-hidden="true" class="components-input-control__backdrop css-173wp57-BackdropUI-rtl em5sgkm6"></div></div></div></div></div><span class="components-datetime__time-separator css-e2elpy-TimeSeparator evcr2315" aria-hidden="true">:</span><div class="components-base-control components-input-control components-number-control components-datetime__time-field-minutes-input evcr2314 ep09it41 css-fk6mz-Wrapper-boxSizingReset-Input-htmlArrowStyles-MinutesInput-baseInput ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><div data-wp-c16t="true" data-wp-component="InputBase" class="components-flex components-input-base em5sgkm5 css-y3fmmm-PolymorphicDiv-Flex-base-ItemsColumn-Root e19lxcc00"><label data-wp-c16t="true" data-wp-component="VisuallyHidden" for="inspector-input-control-1" class="components-visually-hidden css-1qfl78b-PolymorphicDiv e19lxcc00" style="border: 0px; clip: rect(1px, 1px, 1px, 1px); clip-path: inset( 50% ); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; word-wrap: normal;">Minutes</label><div class="components-input-control__container css-7xxir9-Container-containerDisabledStyles-containerWidthStyles em5sgkm4"><input autocomplete="off" inputmode="numeric" max="59" min="0" required="" step="1" class="components-input-control__input css-1chjf1n-Input-dragStyles-fontSizeStyles-sizeStyles-customPaddings em5sgkm3" id="inspector-input-control-1" type="number" value="00"><div aria-hidden="true" class="components-input-control__backdrop css-173wp57-BackdropUI-rtl em5sgkm6"></div></div></div></div></div></div><div class="components-base-control css-fothu4-Wrapper-boxSizingReset ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><div role="radiogroup" aria-label="Select AM or PM" data-wp-c16t="true" data-wp-component="ToggleGroupControl" class="components-toggle-group-control css-x346dc-PolymorphicDiv-__unstable-large-enclosingBorder-enclosingBorders-toggleGroupControl-block e19lxcc00" id="toggle-group-control-as-radio-group-0"><div class="css-1v7jn7q-LabelView-labelBlock et6ln9s1"><button type="button" data-wp-c16t="true" data-wp-component="ToggleGroupControlOptionBase" aria-label="AM" id="toggle-group-control-as-radio-group-0-0" class="css-1ftahvw-buttonView components-toggle-group-control-option-base" data-value="AM" aria-checked="false" role="radio" value="AM"><div class="css-1bj4ucx-ButtonContentView et6ln9s0">AM</div></button></div><div class="css-1v7jn7q-LabelView-labelBlock et6ln9s1"><button type="button" data-wp-c16t="true" data-wp-component="ToggleGroupControlOptionBase" aria-label="PM" id="toggle-group-control-as-radio-group-0-1" class="css-jmuq7p-pressed-buttonView components-toggle-group-control-option-base" data-value="PM" aria-checked="true" data-active-item="true" role="radio" value="PM"><div class="css-1bj4ucx-ButtonContentView et6ln9s0">PM</div></button></div></div></div></div></div><div data-wp-c16t="true" data-wp-component="FlexItem" class="components-flex-item css-1mqm1jz-PolymorphicDiv-Item-sx-Base-block e19lxcc00"></div><div class="components-datetime__timezone css-kd7hyb-TimeZone evcr2310">UTC+0</div></div></fieldset><fieldset class="css-1jlauhr-Fieldset evcr2318"><legend class="components-base-control__label components-datetime__time-legend css-1w2n0ib-StyledVisualLabel-baseLabelTypography-labelStyles ej5x27r0">Date</legend><div data-wp-c16t="true" data-wp-component="HStack" class="components-flex components-h-stack components-datetime__time-wrapper css-1897a1g-PolymorphicDiv-Flex-base-ItemsRow e19lxcc00"><div class="css-17js4ya-MonthSelectWrapper evcr2313"><div class="components-base-control css-fothu4-Wrapper-boxSizingReset ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><div data-wp-c16t="true" data-wp-component="InputBase" class="components-flex components-input-base components-select-control components-datetime__time-field components-datetime__time-field-month e1mv6sxx3 em5sgkm5 css-1r4j154-PolymorphicDiv-Flex-base-ItemsColumn-Root-StyledInputBase e19lxcc00"><label data-wp-c16t="true" data-wp-component="VisuallyHidden" for="inspector-select-control-0" class="components-visually-hidden css-1qfl78b-PolymorphicDiv e19lxcc00" style="border: 0px; clip: rect(1px, 1px, 1px, 1px); clip-path: inset( 50% ); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; word-wrap: normal;">Month</label><div class="components-input-control__container css-7xxir9-Container-containerDisabledStyles-containerWidthStyles em5sgkm4"><select class="components-select-control__input css-j6wvxe-Select-fontSizeStyles-sizeStyles-rtl e1mv6sxx2" id="inspector-select-control-0"><option value="01">January</option><option value="02">February</option><option value="03">March</option><option value="04">April</option><option value="05">May</option><option value="06">June</option><option value="07">July</option><option value="08">August</option><option value="09">September</option><option value="10">October</option><option value="11">November</option><option value="12">December</option></select><span class="components-input-control__suffix css-oobb1x-Suffix em5sgkm7"><div data-wp-c16t="true" data-wp-component="InputControlSuffixWrapper" class="components-input-control-suffix-wrapper e1mv6sxx0 css-1so2tqs-PrefixSuffixWrapper-prefixSuffixWrapperStyles-InputControlSuffixWrapperWithClickThrough-rtl em5sgkm0"><div class="css-di1hl1-DownArrowWrapper e1mv6sxx1"><svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" width="18" height="18" aria-hidden="true" focusable="false"><path d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"></path></svg></div></div></span><div aria-hidden="true" class="components-input-control__backdrop css-173wp57-BackdropUI-rtl em5sgkm6"></div></div></div></div></div></div><div class="components-base-control components-input-control components-number-control components-datetime__time-field components-datetime__time-field-day evcr2312 ep09it41 css-cnot9x-Wrapper-boxSizingReset-Input-htmlArrowStyles-DayInput-baseInput ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><div data-wp-c16t="true" data-wp-component="InputBase" class="components-flex components-input-base em5sgkm5 css-y3fmmm-PolymorphicDiv-Flex-base-ItemsColumn-Root e19lxcc00"><label data-wp-c16t="true" data-wp-component="VisuallyHidden" for="inspector-input-control-2" class="components-visually-hidden css-1qfl78b-PolymorphicDiv e19lxcc00" style="border: 0px; clip: rect(1px, 1px, 1px, 1px); clip-path: inset( 50% ); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; word-wrap: normal;">Day</label><div class="components-input-control__container css-7xxir9-Container-containerDisabledStyles-containerWidthStyles em5sgkm4"><input autocomplete="off" inputmode="numeric" max="31" min="1" required="" step="1" class="components-input-control__input css-1chjf1n-Input-dragStyles-fontSizeStyles-sizeStyles-customPaddings em5sgkm3" id="inspector-input-control-2" type="number" value="18"><div aria-hidden="true" class="components-input-control__backdrop css-173wp57-BackdropUI-rtl em5sgkm6"></div></div></div></div></div><div class="components-base-control components-input-control components-number-control components-datetime__time-field components-datetime__time-field-year evcr2311 ep09it41 css-oepp35-Wrapper-boxSizingReset-Input-htmlArrowStyles-YearInput-baseInput ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><div data-wp-c16t="true" data-wp-component="InputBase" class="components-flex components-input-base em5sgkm5 css-y3fmmm-PolymorphicDiv-Flex-base-ItemsColumn-Root e19lxcc00"><label data-wp-c16t="true" data-wp-component="VisuallyHidden" for="inspector-input-control-3" class="components-visually-hidden css-1qfl78b-PolymorphicDiv e19lxcc00" style="border: 0px; clip: rect(1px, 1px, 1px, 1px); clip-path: inset( 50% ); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; word-wrap: normal;">Year</label><div class="components-input-control__container css-7xxir9-Container-containerDisabledStyles-containerWidthStyles em5sgkm4"><input autocomplete="off" inputmode="numeric" max="9999" min="1" required="" step="1" class="components-input-control__input css-1chjf1n-Input-dragStyles-fontSizeStyles-sizeStyles-customPaddings em5sgkm3" id="inspector-input-control-3" type="number" value="1986"><div aria-hidden="true" class="components-input-control__backdrop css-173wp57-BackdropUI-rtl em5sgkm6"></div></div></div></div></div></div></fieldset></div><div class="components-datetime__date css-1dybc0x-Wrapper-boxSizingReset e105ri6r5" role="application" aria-label="Calendar"><div data-wp-c16t="true" data-wp-component="HStack" class="components-flex components-h-stack e105ri6r4 css-1ept3a2-PolymorphicDiv-Flex-base-ItemsRow-Navigator e19lxcc00"><button type="button" aria-label="View previous month" class="components-button is-compact is-tertiary has-icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24" aria-hidden="true" focusable="false"><path d="M20 11.2H6.8l3.7-3.7-1-1L3.9 12l5.6 5.5 1-1-3.7-3.7H20z"></path></svg></button><h3 data-wp-c16t="true" data-wp-component="Heading" class="components-truncate components-text components-heading e105ri6r3 css-1nv6238-PolymorphicDiv-Text-sx-Base-block-NavigatorHeading e19lxcc00"><strong>October</strong> 1986</h3><button type="button" aria-label="View next month" class="components-button is-compact is-tertiary has-icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24" aria-hidden="true" focusable="false"><path d="m14.5 6.5-1 1 3.7 3.7H4v1.6h13.2l-3.7 3.7 1 1 5.6-5.5z"></path></svg></button></div><div class="css-14z29s7-Calendar e105ri6r2"><div class="css-1tx0y88-DayOfWeek e105ri6r1">Sun</div><div class="css-1tx0y88-DayOfWeek e105ri6r1">Mon</div><div class="css-1tx0y88-DayOfWeek e105ri6r1">Tue</div><div class="css-1tx0y88-DayOfWeek e105ri6r1">Wed</div><div class="css-1tx0y88-DayOfWeek e105ri6r1">Thu</div><div class="css-1tx0y88-DayOfWeek e105ri6r1">Fri</div><div class="css-1tx0y88-DayOfWeek e105ri6r1">Sat</div><button type="button" tabindex="-1" aria-label="October 1, 1986" class="components-button components-datetime__date__day css-h76uxa-DayButton e105ri6r0">1</button><button type="button" tabindex="-1" aria-label="October 2, 1986" class="components-button components-datetime__date__day css-kcybuy-DayButton e105ri6r0">2</button><button type="button" tabindex="-1" aria-label="October 3, 1986" class="components-button components-datetime__date__day css-1ekbkh9-DayButton e105ri6r0">3</button><button type="button" tabindex="-1" aria-label="October 4, 1986" class="components-button components-datetime__date__day css-kfbqby-DayButton e105ri6r0">4</button><button type="button" tabindex="-1" aria-label="October 5, 1986" class="components-button components-datetime__date__day css-vl79ca-DayButton e105ri6r0">5</button><button type="button" tabindex="-1" aria-label="October 6, 1986" class="components-button components-datetime__date__day css-hxvtua-DayButton e105ri6r0">6</button><button type="button" tabindex="-1" aria-label="October 7, 1986" class="components-button components-datetime__date__day css-lrmwlm-DayButton e105ri6r0">7</button><button type="button" tabindex="-1" aria-label="October 8, 1986" class="components-button components-datetime__date__day css-h76uxa-DayButton e105ri6r0">8</button><button type="button" tabindex="-1" aria-label="October 9, 1986" class="components-button components-datetime__date__day css-kcybuy-DayButton e105ri6r0">9</button><button type="button" tabindex="-1" aria-label="October 10, 1986" class="components-button components-datetime__date__day css-1ekbkh9-DayButton e105ri6r0">10</button><button type="button" tabindex="-1" aria-label="October 11, 1986" class="components-button components-datetime__date__day css-kfbqby-DayButton e105ri6r0">11</button><button type="button" tabindex="-1" aria-label="October 12, 1986" class="components-button components-datetime__date__day css-vl79ca-DayButton e105ri6r0">12</button><button type="button" tabindex="-1" aria-label="October 13, 1986" class="components-button components-datetime__date__day css-hxvtua-DayButton e105ri6r0">13</button><button type="button" tabindex="-1" aria-label="October 14, 1986" class="components-button components-datetime__date__day css-lrmwlm-DayButton e105ri6r0">14</button><button type="button" tabindex="-1" aria-label="October 15, 1986" class="components-button components-datetime__date__day css-h76uxa-DayButton e105ri6r0">15</button><button type="button" tabindex="-1" aria-label="October 16, 1986" class="components-button components-datetime__date__day css-kcybuy-DayButton e105ri6r0">16</button><button type="button" tabindex="-1" aria-label="October 17, 1986" class="components-button components-datetime__date__day css-1ekbkh9-DayButton e105ri6r0">17</button><button type="button" tabindex="0" aria-label="October 18, 1986. Selected" class="components-button components-datetime__date__day css-16e7a4x-DayButton e105ri6r0">18</button><button type="button" tabindex="-1" aria-label="October 19, 1986" class="components-button components-datetime__date__day css-vl79ca-DayButton e105ri6r0">19</button><button type="button" tabindex="-1" aria-label="October 20, 1986" class="components-button components-datetime__date__day css-hxvtua-DayButton e105ri6r0">20</button><button type="button" tabindex="-1" aria-label="October 21, 1986" class="components-button components-datetime__date__day css-lrmwlm-DayButton e105ri6r0">21</button><button type="button" tabindex="-1" aria-label="October 22, 1986" class="components-button components-datetime__date__day css-h76uxa-DayButton e105ri6r0">22</button><button type="button" tabindex="-1" aria-label="October 23, 1986" class="components-button components-datetime__date__day css-kcybuy-DayButton e105ri6r0">23</button><button type="button" tabindex="-1" aria-label="October 24, 1986" class="components-button components-datetime__date__day css-1ekbkh9-DayButton e105ri6r0">24</button><button type="button" tabindex="-1" aria-label="October 25, 1986" class="components-button components-datetime__date__day css-kfbqby-DayButton e105ri6r0">25</button><button type="button" tabindex="-1" aria-label="October 26, 1986" class="components-button components-datetime__date__day css-vl79ca-DayButton e105ri6r0">26</button><button type="button" tabindex="-1" aria-label="October 27, 1986" class="components-button components-datetime__date__day css-hxvtua-DayButton e105ri6r0">27</button><button type="button" tabindex="-1" aria-label="October 28, 1986" class="components-button components-datetime__date__day css-lrmwlm-DayButton e105ri6r0">28</button><button type="button" tabindex="-1" aria-label="October 29, 1986" class="components-button components-datetime__date__day css-h76uxa-DayButton e105ri6r0">29</button><button type="button" tabindex="-1" aria-label="October 30, 1986" class="components-button components-datetime__date__day css-kcybuy-DayButton e105ri6r0">30</button><button type="button" tabindex="-1" aria-label="October 31, 1986" class="components-button components-datetime__date__day css-1ekbkh9-DayButton e105ri6r0">31</button></div></div></div></div></div></div><div class="components-panel__row pods-inspector-row"><div class="components-base-control css-fothu4-Wrapper-boxSizingReset ej5x27r4"><div class="components-base-control__field css-2txpkp-StyledField ej5x27r3"><label class="components-base-control__label css-1jylnrs-StyledLabel-baseLabelTypography-labelStyles ej5x27r2" for="inspector-number-control-0">Label for the number field</label><input inputmode="numeric" id="inspector-number-control-0" class="component-number-control" type="number" value="123456"></div></div></div>"`;
